lib/utilities/get-config.js

```diff
@@ -1,23 +0,0 @@
-'use strict';
-
-let config;
-let Yam = require('yam');
-let Project = require('../models/project');
-
-function generateConfig() {
-  return new Yam('ember-cli', {
-    primary: Project.getProjectRoot(),
-  });
-}
-
-module.exports = function getConfig(configOverride) {
-  if (configOverride) {
-    return configOverride;
-  }
-
-  if (config === undefined) {
```

Comment:
I'm happy to remove caching, I was being conservative as I'm not familiar with the impact of removing it. I'll prepare a separate PR while we consider

Reply:
This function essentially becomes:

```js
function getConfig(override) {
  if (override) { return override; }

  return generateConfig();
}
```

Reply:
Sounds good. FWIW, this caching should _only_ affect the blueprint testing case. In "normal" command line usage, we would only ever call `getConfig` once anyways...

Reply:
It's also called in [`instrumentationEnabled`](https://github.com/ember-cli/ember-cli/blob/252e1ccd1a99222397c8c9b293eb5e5f376070e1/lib/utilities/instrumentation.js#L25)

Reply:
Thanks for sorting @GavinJoyce 👏

@rwjblue I was under the impression that it would have been called quite frequently in the `instrumentation` checks but I could be wrong. Is it just called on instrumentation set up?

Reply:
Ya, its called 3 times